packages/core/src/repository/data-source/in-memory.data-source.ts

```diff
@@ -49,7 +50,11 @@ export class InMemoryDataSource<T> implements GetDataSource<T>, PutDataSource<T>
         }
     }
 
-    public async put(value: T, query: Query): Promise<T> {
+    public async put(value: T | undefined, query: Query): Promise<T> {
+        if (typeof value === 'undefined') {
```

Comment:
Shouldn't the delete method be used for that, @orioljp ?

Reply:
We require `value` in the InMemoryDataSource as we do not want to allow `null` properties in our array. 

Could Undefined mean `delete`? Yes, it can, but we should throw the error, and the developer must use the API correctly.